nengo/connection.py

```diff
@@ -233,9 +272,12 @@ class Connection(NengoObject):
     synapse : Synapse, optional \
               (Default: ``nengo.synapses.Lowpass(tau=0.005)``)
         Synapse model to use for filtering (see `~nengo.synapses.Synapse`).
-    function : callable, optional (Default: None)
+    function : callable or (n_eval_points, size_mid) array_like, \
+               optional (Default: None)
         Function to compute across the connection. Note that ``pre`` must be
         an ensemble to apply a function across the connection.
+        If an array is passed, the function is implicitly defined by the
+        array points and the provided ``eval_points``.
```

Comment:
It gives the required shape above. Does that make it clear that it's 2D?

If I read "array points referenced" I'm not sure I would think that means they're passed by reference, i.e. not copied. Also, I think the assumption is that anything passed as a parameter is not copied, so if you pass an array and then change a value it will change that in your model. Of course, when everything is built, we make copies as needed so that the built model would never be changed.